### PR TITLE
Build - Tools - Add Oracles

### PR DESCRIPTION
**Pull request by lucas-tortora**
# Description of change

Added article about oracles, including Supra and Pyth.

## Links to any relevant issues

fixes https://github.com/iotaledger/iota-wiki/issues/1407

## Type of change

- Documentation 

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the [contribution guidelines](https://github.com/iota-wiki/iota-wiki/blob/main/.github/CONTRIBUTING.md) for this project
- [x] I have performed a self-review of my own changes
- [x] I have made sure that added/changed links still work
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by lucas-tortora** [comment]
https://iota-wiki-itjzqm8b6-iota1.vercel.app/build/oracles/

File: docs/build/getting-started/oracles.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+async function main() {
+  try {
+    await getPrice();
+  } catch (error) {
+    console.error('Main Error:', error.message);
+  }
+}
+
+main();
+```
```
Comment (Dr-Electron) [inline docs/build/getting-started/oracles.md:129]:
```suggestion

```typescript
import {ethers} from "ethers";
import fetch from "node-fetch";

// Provider
const providerURL = "https://json-rpc.evm.shimmer.network"
const provider = new ethers.JsonRpcProvider(providerURL);

//  Pyth Oracle contract on ShimmerEVM Mainnet
const contractAddress = '0xA2aa501b19aff244D90cc15a4Cf739D2725B5729';

// Add Price Feed Ids obtained from https://pyth.network/developers/price-feed-ids#pyth-evm-stable
const priceFeedIds = [
    //SMR/USD
    '0xaf5b9ac426ae79591fde6816bc3f043b5e06d5e442f52112f76249320df22449',
    //BTC/USD
    '0xe62df6c8b4a85fe1a67db44dc12de5db330f7ac66b72dc658afedf0f4a415b43'
]

async function getPrice() {
  // Replace 'IPythJSONURL' with the actual URL where the IPyth ABI JSON is hosted
  const pythAbiUrl = 'https://raw.githubusercontent.com/pyth-network/pyth-sdk-solidity/main/abis/IPyth.json';
  const pythAbi = await fetch(pythAbiUrl).then((response) => response.json());

  const pythContract = new ethers.Contract(contractAddress, pythAbi, provider);

   try {
    for (const priceFeedId of priceFeedIds) {
      const priceData = await pythContract.getPrice(priceFeedId);
      const price = Number(priceData.price) * 10 ** (Number(priceData.expo));

      console.log(`Price for ${priceFeedId}:`);
      console.log('Price:', price);
      console.log('Confidence Interval:', priceData.conf.toString());
      // Convert publishTime to a formatted date
      const publishTimeDate = new Date(number(priceData.publishTime) * 1000);
      console.log('Publish Time:', publishTimeDate.toUTCString());
      console.log('\n');
    }
  } catch (error) {
    console.error('Error:', error.message);
  }
}

async function main() {
  try {
    await getPrice();
  } catch (error) {
    console.error('Main Error:', error.message);
  }
}

main();
```
```

Comment (Dr-Electron) [reply docs/build/getting-started/oracles.md:129]: This works for me with ethers v6

**Review by Phyloiota — APPROVED** [review]
looks good

**Pull request merged**